server/base/src/main/java/org/apache/accumulo/server/metrics/MetricsInfoImpl.java

```diff
@@ -0,0 +1,320 @@
+        new ClassLoaderMetrics().bindTo(composite);
+        new JvmMemoryMetrics().bindTo(composite);
+        jvmGcMetrics = new JvmGcMetrics();
+        jvmGcMetrics.bindTo(composite);
+        new ProcessorMetrics().bindTo(composite);
+        new JvmThreadMetrics().bindTo(composite);
+      }
+
+      MeterFilter replicationFilter = new MeterFilter() {
+        @Override
+        public DistributionStatisticConfig configure(Meter.Id id,
+            @NonNull DistributionStatisticConfig config) {
+          if (id.getName().equals("replicationQueue")) {
+            return DistributionStatisticConfig.builder().percentiles(0.5, 0.75, 0.9, 0.95, 0.99)
+                .expiry(Duration.ofMinutes(10)).build().merge(config);
+          }
+          return config;
+        }
+      };
+
+      if (isMetricsEnabled()) {
+        // user specified registries
+        String userRegistryFactories =
+            context.getConfiguration().get(Property.GENERAL_MICROMETER_FACTORY);
+
+        for (String factoryName : getTrimmedStrings(userRegistryFactories)) {
+          try {
+            MeterRegistry registry = getRegistryFromFactory(factoryName, context);
+            registry.config().commonTags(commonTags.values());
+            registry.config().meterFilter(replicationFilter);
+            addRegistry(registry);
+          } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException
+              | InstantiationException | IllegalAccessException ex) {
+            LOG.warn("Could not load registry {}", factoryName, ex);
+          }
+        }
+      }
+
+      pendingRegistries.forEach(registry -> composite.add(registry));
+
+      LOG.info("Metrics initialization. Register producers: {}", producers);
+      producers.forEach(p -> p.registerMetrics(composite));
+
+      Metrics.globalRegistry.add(composite);
+
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @VisibleForTesting
+  @SuppressWarnings({"deprecation",
+      "support for org.apache.accumulo.core.metrics.MeterRegistryFactory can be removed in 3.1"})
+  static MeterRegistry getRegistryFromFactory(final String factoryName, final ServerContext context)
+      throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+      InstantiationException, IllegalAccessException {
+    try {
+      LOG.info("look for meter spi registry factory {}", factoryName);
+      Class<? extends org.apache.accumulo.core.spi.metrics.MeterRegistryFactory> clazz =
+          ClassLoaderUtil.loadClass(factoryName,
+              org.apache.accumulo.core.spi.metrics.MeterRegistryFactory.class);
+      org.apache.accumulo.core.spi.metrics.MeterRegistryFactory factory =
+          clazz.getDeclaredConstructor().newInstance();
+      org.apache.accumulo.core.spi.metrics.MeterRegistryFactory.InitParameters initParameters =
+          new MeterRegistryEnvPropImpl(context);
+      factory.setInitParams(initParameters);
+      return factory.create();
```

Comment:
Is there a reason to have two different methods here? Is there a case where `setInitParams` is called separately from `create`?

Reply:
I was leaking implementation details into the SPI interface.  7d9740cd60 removes the setInitParams method as well as the synchronization and locking.

The separation of the SPI and the implementation in MeterRegistryEnvPropImpl should be better defined with these changes,